stable/harbor/questions.yaml

```diff
@@ -0,0 +1,577 @@
+- variable: externalURL
+  default: "https://harbor.domain"
+  description: "Set the external URL used to access Harbor. If exposed using an Ingress, it should reflect the Ingress Hostname used for Harbor (format: protocol://domain:port)"
+  type: string
+  label: The External URL for Harbor Service
+  group: "Expose Service Options"
+- variable: expose.loadBalancer.IP
+  default: ""
+  description: "Set the IP if the LoadBalancer supports assigning IP"
+  type: string
+  label: Set LoadBalancer IP
+  show_if: "expose.type=loadBalancer"
+  group: "Expose Service Options"
+- variable: expose.tls.certSource
+  required: true
+  default: "auto"
+  description: "The source for the Harbor TLS certificate"
+  type: enum
+  label: Choose TLS Certificate Source
+  group: "Expose TLS Options"
+  options:
+  - "auto"
+  - "secret"
+  - "none"
+- variable: expose.tls.commonName
+  default: ""
+  required: true
+  description: "The common name used to generate the certificate"
+  type: string
+  label: Common Name for the TLS Certificate
+  show_if: "expose.tls.certSource=auto&&expose.type!=ingress"
+  group: "Expose TLS Options"
+- variable: expose.tls.auto.rotateCert
+  default: false
+  description: "Generate new TLS certificates during post-deployment configuration changes and upgrades"
+  type: boolean
+  label: Rotate TLS Certificates
+  show_if: "expose.tls.certSource=auto"
+  group: "Expose TLS Options"
+- variable: expose.tls.secret.secretName
+  default: ""
+  required: true
+  description: "The name of the Harbor TLS certificate secret. The secret must contain keys named tls.crt and tls.key representing the certificate and private key to use for TLS"
+  type: certificate
+  label: Select a Secret Holding the Harbor TLS Certificate
+  show_if: "expose.tls.certSource=secret"
+  group: "Expose TLS Options"
+- variable: expose.tls.secret.notarySecretName
+  default: ""
+  required: true
+  description: "The name of the Notary TLS certificate secret. The secret must contain keys named tls.crt and tls.key representing the certificate and private key to use for TLS"
+  type: certificate
+  label: Select a Secret Holding the Notary TLS Certificate
+  show_if: "expose.tls.certSource=secret&&expose.type=ingress"
+  group: "Expose TLS Options"
+- variable: internalTLS.enable
+  default: false
+  required: true
+  description: "Use TLS to secure internal communication between Harbor components"
+  type: boolean
+  label: Enable Internal TLS
+  group: "Internal TLS Options"
+- variable: internalTLS.rotateCert
+  default: false
+  description: "Generate new TLS certificates during post-deployment configuration changes and upgrades"
+  type: boolean
+  label: Rotate TLS Certificates
+  show_if: "internalTLS.enable=true"
+  group: "Internal TLS Options"
+# config imageChartStorage type
+- variable: persistence.imageChartStorage.type
+  default: "filesystem"
+  description: "Specify the type of storage for OCI artifacts: filesystem, azure, gcs, s3, swift, oss"
```

Comment:
Hm, I'm not sure if it makes sense to just list the values here. What could be useful is to state that for filesystem type, default storageClass will be used.

Reply:
You're right, it makes no sense to just enumerate the values. I changed the description to point out the difference between `filesystem` and the other options.